sign_and_install.sh

```diff
@@ -14,28 +15,22 @@ then error "Please run as root"
 fi
 
 dkms_mok_dir="${dkms_mok_dir:-/var/lib/shim-signed/mok}"
-mokutil_509key_pass=$(cat "${dkms_mok_dir}/.openssl_pass" 2> /dev/null)
-mokutil_509key_pass="${mokutil_509key_pass:-$(openssl rand -hex 6)}"
 
 command -v openssl > /dev/null 2>&1 || ( warn "missing openssl, you will not be able to create a new MOK key")
 command -v mokutil > /dev/null 2>&1 || ( warn "missing mokutil, you will not be able to sign the module")
-ls /var/lib/shim-signed > /dev/null 2>&1 || ( warn "missing shim-signed, you will not be able to sign the module with dkms")
+ls /var/lib/shim-signed > /dev/null 2>&1 || ( error "missing shim-signed, unable to enrol MOK")
```

Comment:
Don't use `ls` to check for presence of a file name. You're expecting either a readable file or a directory. Use `[ -d /var/lib/shim-signed ] || error "missing ..."` or if you want to support symlinks, use `[ -d /var/lib/shim-signed/. ] || ...`. See `man test` for more options. Also, is it necessary to spawn `( subshells )` here?